examples/01_plotting/plot_surf_stat_map.py

```diff
@@ -83,6 +83,7 @@
 from nilearn import surface
 
 timeseries = surface.load_surf_data(nki_dataset["func_left"][0])
+timeseries = timeseries.astype(float)
```

Comment:
this fixes the issue but that feels very hacky to me

wouldn't it be better if we ensured that load_surf_data returns a consistent dtype?

@bthirion @jeromedockes

Reply:
not `load_surf_data` -- surface data can have any type eg atlases will be integers, statistics will be floating-point. But the cast can be done inside of the NKI fetcher so that the time series are floats

Reply:
That could work for the experimental the new experimental NKI fetchers for the SurfaceImage object as we have to load the data anyway.

But the regular NKI fetcher just returns Paths... So loading and resaving the data in a proper format sounds not ideal.

I think for this PR I am just going to leave it at that and add some comments, so that at least the doc build is fixed and then open a new separate issue to discuss how to cleanly handle this.